leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -167,11 +169,25 @@ public Boolean call() throws Exception {
             Future<Boolean> pingFuture = pingCompletionService.poll(
                     leaderPingResponseWaitMs,
                     TimeUnit.MILLISECONDS);
-            return pingFuture != null && pingFuture.get();
+            if (pingFuture != null) {
+                if (pingFuture.get()) {
+                    leaderLog.trace("Successfully pinged " + leaderName);
+                        return true;
+                } else {
+                    leaderLog.info("Pinged " + leaderName + " and it reported that it is no longer the leader");
+                    return false;
+                }
+            } else {
+                leaderLog.warn("Timed out pinging " + leaderName + " after " + leaderPingResponseWaitMs + "ms");
+                return false;
+            }
         } catch (InterruptedException e) {
+            leaderLog.warn("Interrupted while pinging " + leaderName);
+            log.warn("Interrupted while pinging " + leaderName);
             return false;
         } catch (ExecutionException e) {
-            log.warn("cannot ping leader", e);
+            log.warn("Cannot ping " + leaderName, e);
```

Comment:
Very pedantic nit: you have your calls to `log` and `leaderLog` in different orders within the two catch blocks.